packages/babel-plugin-transform-for-of/src/index.ts

```diff
@@ -234,10 +242,19 @@ export default declare((api, options: Options) => {
           );
         }
 
-        // ensure that it's a block so we can take all its statements
-        path.ensureBlock();
-
-        (node.body as t.BlockStatement).body.unshift(declar);
+        let blockBody;
+        const body = path.get("body");
+        if (
+          body.isBlockStatement() &&
+          Object.keys(path.getBindingIdentifiers()).some(id =>
+            body.scope.hasOwnBinding(id),
+          )
```

Comment:
For further PRs: this check is similar to
https://github.com/babel/babel/blob/cc2de3b7ee85d7bb52ebfae799f1a1c6af2ad8f6/packages/babel-plugin-transform-destructuring/src/util.ts#L12-L15

Maybe we can extract them into a new helper.

Reply:
100% agree, I was thinking the same thing while reviewing.